balancesExporter/blocks/blocksRepository.go

```diff
@@ -124,24 +129,18 @@ func (repository *blocksRepository) loadMarshalizedBlocksInEpoch() ([][]byte, er
 }
 
 func (repository *blocksRepository) getStorageUnitPath() string {
-	epochPart := fmt.Sprintf("Epoch_%d", repository.epoch)
-	shardPart := fmt.Sprintf("Shard_%d", repository.shard)
-	unitPath := path.Join(repository.dbPath, epochPart, shardPart, "BlockHeaders")
-	return unitPath
+	pathManager := common.NewSimplePathManager(repository.dbPath)
+	path := pathManager.PathForEpoch(strconv.Itoa(int(repository.shard)), repository.epoch, storageUnitIdentifier)
+	return path
 }
 
 func hasScheduledMiniblocks(block data.HeaderHandler) bool {
 	miniblocks := block.GetMiniBlockHeaderHandlers()
 
 	for _, miniblock := range miniblocks {
 		processingType := dataBlock.ProcessingType(miniblock.GetProcessingType())
-
-		if processingType == dataBlock.Scheduled {
-			return true
-		}
-		if processingType == dataBlock.Processed {
-			return true
-		}
+		isScheduledProcessingType := processingType == dataBlock.Scheduled || processingType == dataBlock.Processed
```

Comment:
This code is not the same. It can return false after the first miniblock it tries to process

Reply:
Thanks for the catch :pray: . Fixed.